index2.html

```diff
@@ -0,0 +1,1260 @@
+    <dt>readonly attribute NFC nfc</dt>
+  </dl>
+  <!-- - - - - - - - - - - - nfc attribute  - - - - - - - - - - - -->
+  <section> <h3>The <strong>nfc</strong> attribute</h3>
+  <p>
+    When getting the <dfn id="widl-Navigator-nfc">nfc</dfn> attribute, the
+    <a>user agent</a> MUST return the <a>NFC</a> object, which provides
+    NFC related functionality.
+  </p>
+  </section> <!-- nfc attribute -->
+</section> <!-- Navigator -->
+
+<section> <h2>The <strong>NFC</strong> interface</h2>
+  <dl title="interface NFC" class="idl">
+    <dt>Promise findAdapters()</dt>
+  </dl>
+  <p>
+    Implementations might expose multiple NFC adapters. By using the
+    <dfn id="widl-NFC-findAdapters-Promise">findAdapters</dfn>() method,
+    <a>web app</a>s can obtain adapter objects providing NFC functionality,
+    in a similar way to that of service worker extensions [[SERVICE-WORKERS]].
+    When this method is invoked, the <a>user agent</a> MUST run the following
+    steps:
+    <ol id="steps-findAdapters">
+      <li>Let <var>promise</var> be a new <a><code>Promise</code></a> object.
+      </li>
+      <li>
+        Return <var>promise</var> and continue the following steps
+        asynchronously.
+      </li>
+      <li>
+        If there are no <a>sufficient permission</a>s to use this method,
+        that is, to use NFC technology from the calling <a>web app</a>,
+        then reject <var>promise</var> with a new
+        <code><a>DOMException</a></code> object whose <var>name</var> property
+        is set to <code>"<a>SecurityError</a>"</code>, and terminate these steps.
+      </li>
+      <li>
+        If there is no support for <a>NFC adapter</a> handling functionality in
+        hardware, software, or due to physical incompatibility,
+        then reject <var>promise</var> with a new
+        <code><a>DOMException</a></code> object whose <var>name</var> property
+        is set to <code>"<a>NotSupportedError</a>"</code>, and terminate
+        these steps.
+      </li>
+      <li>
+        Make a request to the underlying platform to initialize NFC
+        functionality and enumerate available adapters. If the request fails,
+        then reject <var>promise</var> with a new
+        <code><a>DOMException</a></code> object whose <var>name</var>
+        property is set to <code>"<a>NotSupportedError</a>"</code>,
+        and terminate these steps.
+      </li>
+      <li>
+        If the request is successful, then resolve <var>promise</var> with an
+        array of <code><a>NfcAdapter</a></code> objects. Implementations SHOULD
+        maintain a default adapter, which SHOULD be exposed as the first
+        element in the returned array.
+      </li>
+    </ol>
+  </p>
+</section> <!-- NFC interface -->
+
+<section> <h2>The <strong>NfcAdapter</strong> interface</h2>
+  <dl title="interface NfcAdapter : EventHandler" class="idl">
+    <dt>attribute EventHandler onread</dt>
+    <dt>attribute EventHandler onbeforepush</dt>
```

Comment:
Based on implementation feedback, I will remove this event, as practically not implementable in a useful way.